### PR TITLE
Updated Cosmos DB docs for Dapr 1.7

### LINKED ISSUE #2325: Cosmos DB state store docs need update with steps for creating stored procedures manually
**What content needs to be created or modified?**

https://github.com/dapr/components-contrib/pull/1613 requires content updates to the page https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-azure-cosmosdb/ to include the manual steps for creating the stored procedures in the database, when using Azure AD to authenticate

**Describe the solution you'd like**
I need to write docs for creating the stored procedures manually

**Where should the new material be placed?**
https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-azure-cosmosdb/

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**
https://github.com/dapr/components-contrib/pull/1613

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
1. More complete instructions for setting up Azure AD auth for the Cosmos DB state store and binding
2. Instructions for setting up the stored procedures manually for the state store

Also includes some general polishing of the docs, including ensuring proper branding("Cosmos DB" and "Azure AD" - and with non-breaking spaces between the words to make them look better!).

Closes #2325

PS: Many thanks to @berndverst for merging the required code changes in components-contib very quickly and allowing these docs :)

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-azure-cosmosdb.md [hunk 1]
```diff
@@ -127,10 +131,102 @@ curl -X POST http://localhost:3500/v1.0/state/<store_name> \
...
+  --role-definition-id "$ROLE_ID"
+```
+
+### Creating the stored procedures for Dapr
+
+When using Cosmos DB as a state store for Dapr, we need to create two stored procedures in your collection. When you configure the state store using a "master key", Dapr creates those for you, automatically. However, when your state store authenticates with Cosmos DB using Azure AD, because of limitations in the platform we are not able to do it automatically.
+
+If you are using Azure AD to authenticate your Cosmos DB state store and have not created the stored procedures (or if you are using an outdated version of them), your Dapr sidecar will fail to start and you will see an error similar to this in your logs:
+
+```text
```
Comment (ItalyPaleAle) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-azure-cosmosdb.md:178]: I am including here the error message in full for SEO. So when people see this error and Google it, they can land on this page.

File: daprdocs/content/en/reference/components-reference/supported-bindings/cosmosdb.md [hunk 2]
```diff
@@ -81,8 +84,47 @@ spec:
...
-- `<partitionKey>`: the name of the partition key specified via the `spec.partitionKey` in the component definition. This must also match the partition key specified upon creation of the Cosmos DB container.
+- `<partitionKey>`: the name of the partition key specified via the `spec.partitionKey` in the component definition. This must also match the partition key specified upon creation of the Cosmos DB container.
+
+## Setting up Cosmos DB for authenticating with Azure AD
+
+When using the Dapr Cosmos DB binding and authenticating with Azure AD, you need to perform a few additional steps to set up your environment.
+
+Prerequisites:
+
+- You will need a Service Principal created as per the instructions in the [authenticating to Azure]({{< ref authenticating-azure.md >}}) page. You will need the ID of the Service Principal for the commands below (note that this is different from the client ID of your application, or the value you use for `azureClientId` in the metadata).
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/cosmosdb.md]:
```suggestion
- You need a Service Principal created as per the instructions in the [authenticating to Azure]({{< ref authenticating-azure.md >}}) page. You need the ID of the Service Principal for the commands below (note that this is different from the client ID of your application, or the value you use for `azureClientId` in the metadata).
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/cosmosdb.md [hunk 3]
```diff
@@ -81,8 +84,47 @@ spec:
...
+- [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli)
+- [jq](https://stedolan.github.io/jq/download/)
+- The scripts below are optimized for a bash or zsh shell
+
+> When using the Cosmos DB binding, you don't need to create stored procedures like in the case of the Cosmos DB state store.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/cosmosdb.md]:
```suggestion
> When using the Cosmos DB binding, you **don't** need to create stored procedures as you do in the case of the Cosmos DB state store.
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-azure-cosmosdb.md [hunk 4]
```diff
@@ -88,17 +92,17 @@ spec:
 
 ## Data format
 
-To use the CosmosDB state store, your data must be sent to Dapr in JSON-serialized.  Having it just JSON *serializable* will not work.
+To use the Cosmos DB state store, your data must be sent to Dapr in JSON-serialized format. Having it just JSON *serializable* will not work.
 
-If you are using the Dapr SDKs (e.g. https://github.com/dapr/dotnet-sdk) the SDK will serialize your data to json.
+If you are using the Dapr SDKs (e.g. the [.NET SDK](https://github.com/dapr/dotnet-sdk)), the SDK will serialize your data to JSON for you, automatically.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-azure-cosmosdb.md]:
```suggestion
If you are using the Dapr SDKs (for example the [.NET SDK](https://github.com/dapr/dotnet-sdk)), the SDK automatically serializes your data to JSON.
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-azure-cosmosdb.md [hunk 5]
```diff
@@ -88,17 +92,17 @@ spec:
...
-For examples see the curl operations in the [Partition keys](#partition-keys) section.
+If you want to invoke Dapr's HTTP endpoint directly, take a look at the examples (using curl) in the [Partition keys](#partition-keys) section below.
 
 ## Partition keys
 
-For **non-actor state** operations, the Azure Cosmos DB state store will use the `key` property provided in the requests to the Dapr API to determine the Cosmos DB partition key.  This can be overridden by specifying a metadata field in the request with a key of `partitionKey` and a value of the desired partition.
+For **non-actor state** operations, the Azure Cosmos DB state store will use the `key` property provided in the requests to the Dapr API to determine the Cosmos DB partition key. This can be overridden by specifying a metadata field in the request with a key of `partitionKey` and a value of the desired partition.
 
-The following operation will use `nihilus` as the partition key value sent to CosmosDB:
+The following operation will use `nihilus` as the partition key value sent to Cosmos DB:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-azure-cosmosdb.md]:
```suggestion
The following operation uses `nihilus` as the partition key value sent to Cosmos DB:
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-azure-cosmosdb.md [hunk 6]
```diff
@@ -127,10 +131,102 @@ curl -X POST http://localhost:3500/v1.0/state/<store_name> \
...
 
+For **actor** state operations, the partition key is generated by Dapr using the `appId`, the actor type, and the actor id, such that data for the same actor always ends up under the same partition (you do not need to specify it). This is because actor state operations must use transactions, and in Cosmos DB the items in a transaction must be on the same partition.
+
+## Setting up Cosmos DB for authenticating with Azure AD
+
+When using the Dapr Cosmos DB state store and authenticating with Azure AD, you need to perform a few additional steps to set up your environment.
+
+Prerequisites:
+
+- You will need a Service Principal created as per the instructions in the [authenticating to Azure]({{< ref authenticating-azure.md >}}) page. You will need the ID of the Service Principal for the commands below (note that this is different from the client ID of your application, or the value you use for `azureClientId` in the metadata).
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-azure-cosmosdb.md]:
```suggestion
- You need a Service Principal created as per the instructions in the [authenticating to Azure]({{< ref authenticating-azure.md >}}) page. You need the ID of the Service Principal for the commands below (note that this is different from the client ID of your application, or the value you use for `azureClientId` in the metadata).
```

**Review by msfussell — APPROVED** [review]
Fantastic update! LGTM

**Comment by ItalyPaleAle** [comment]
Thanks for the review @msfussell ! Looks great to me, #ShipIt!

**Pull request merged**